### PR TITLE
[FEAT] Add friction to SRP reveal

### PR DESCRIPTION
**Pull request by gantunesr**
**Description**

Introduce friction to the SRP reveal flow to reduce the phishing attempts again our users.

**Screenshots/Recordings**

- Android: https://github.com/MetaMask/mobile-planning/issues/591#issuecomment-1400841241
- iOS: https://github.com/MetaMask/mobile-planning/issues/591#issuecomment-1400856439

**Issue**

Progresses https://github.com/MetaMask/mobile-planning/issues/591

**Checklist**

* [x] There is a related GitHub issue
* [ ] Tests are included if applicable
* [x] Any added code is fully documented

### CODE REVIEW

**Review by gantunesr — COMMENTED** [review]
To Do:

- [x] Revert version and build number [b9833bb](https://github.com/MetaMask/metamask-mobile/pull/5579/commits/b9833bb2db5430d99985c3db269cbc30b695746f)
- [x] Revert .iyarc file changes [17d68eb](https://github.com/MetaMask/metamask-mobile/pull/5579/commits/17d68ebf7ba9b684425226eae3225e2c8631e25e)

**Review by sethkfman — COMMENTED** [review]
Remove linter comments and consider other changes.

File: app/components/Views/SRPQuiz/QuizModal/QuizModal.tsx [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+  const wrongAnswerIcon = useCallback(
+    () => (
+      <Icon
+        size={IconSize.Xl}
+        name={IconName.DangerFilled}
+        color={colors.error.default}
+      />
+    ),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [],
```
[the remark this answers is not in the document]
Comment (gantunesr) [reply app/components/Views/SRPQuiz/QuizModal/QuizModal.tsx]: [727cb63](https://github.com/MetaMask/metamask-mobile/pull/5579/commits/727cb63e173c68a04130b07cfc7aaf26e2df29e7)

File: app/components/Views/SRPQuiz/QuizModal/QuizModal.tsx [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+        name={IconName.CheckCircleOnFilled}
+        color={colors.success.default}
+      />
+    ),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [],
+  );
+
+  const goToRevealPrivateCredential = () => {
+    navigation.navigate('RevealPrivateCredentialView', {
```
[the remark this answers is not in the document]
Comment (gantunesr) [reply app/components/Views/SRPQuiz/QuizModal/QuizModal.tsx]: [894467d](https://github.com/MetaMask/metamask-mobile/pull/5579/commits/894467d23b1777453639ee914549a9e5dd5b5d5b)

File: app/components/Views/SRPQuiz/types.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    variant: ButtonVariants;
+  }[];
+  dismiss: () => void;
+  header?: string;
+  image?: ImageSourcePropType;
+  content?: string;
+  icon?: any;
+}
+
+export interface Question {
```
[the remark this answers is not in the document]
Comment (gantunesr) [reply app/components/Views/SRPQuiz/types.ts]: [e0e99fb](https://github.com/MetaMask/metamask-mobile/pull/5579/commits/e0e99fb47f006e084d340c6daf0bd175a3b16f0d)

File: app/components/Views/SRPQuiz/QuizContent/QuizContent.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  title,
+  content,
+  icon,
+  buttons,
+  dismiss,
+}: IQuizInformationProps) => {
+  const { styles, theme } = useStyles(stylesheet, {});
+  const { colors } = theme;
+
+  const openSupportArticle = () => {
```
[the remark this answers is not in the document]
Comment (gantunesr) [reply app/components/Views/SRPQuiz/QuizContent/QuizContent.tsx]: [dce420c](https://github.com/MetaMask/metamask-mobile/pull/5579/commits/dce420c2f00c7a0e069e29db2a2e55d2e18234a3)

File: app/components/Views/SRPQuiz/QuizModal/QuizModal.tsx [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+                onPress: () => setStage(QuizStage.questionTwo),
+                variant: ButtonVariants.Primary,
+              },
+            ]}
+            dismiss={dismissModal}
+          />
+        );
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [navigation, stage, rightAnswerIcon, wrongAnswerIcon]);
```
[the remark this answers is not in the document]
Comment (gantunesr) [reply app/components/Views/SRPQuiz/QuizModal/QuizModal.tsx]: [9632f3a](https://github.com/MetaMask/metamask-mobile/pull/5579/commits/9632f3ae992a40216f569670966cb3d4d8034c2f)

**Review by sethkfman — APPROVED** [review]
LGTM

**Comment by olenapankina** [comment]
**Tested on**
iPhone Xr iOS 16.1.1
Simulator iPhone 12 mini iOS 14.5
Samsung Galaxy M21 Android 12, One UI Core version 4.1
commit ba76f20aa7e08c54e8b0ed370e0d46519d14b5fa

e2e tests [passed](https://app.bitrise.io/build/f18c58cb-e911-411a-8075-0afc7bd80ac1?tab=log)
Feature file [created](https://www.notion.so/Quiz-Add-Friction-to-SRP-Reveal-9361faa733ad45f58d13555d20c1a43c)
QA Passed ✅

**Pull request merged**